spikes/cluster-add-ons/single-app-of-apps-per-cluster/readme.md

```diff
@@ -0,0 +1,170 @@
+    # Wait until all pods are showing 1/1 in ready state
+    kubectl wait pods -n argocd --all --for condition=ready --insecure-skip-tls-verify
+    ```
+
+5. Expose API Server External to Cluster (run this command in a new zsh terminal so port forwarding remains running)
+
+    ``` bash
+    # Forward port to access UI outside of cluster
+    export KUBECONFIG=config-argo
+    kubectl port-forward svc/argocd-server -n argocd 8080:443  --insecure-skip-tls-verify
+    ```
+
+    After this step is complete go back to original terminal to run the rest of the commands
+
+6. Access UI
+
+    1. Get initial password
+
+        ``` bash
+        # Get the initial password for installation - make note
+        argocd admin initial-password -n argocd
+        ````
+
+    2. You can now access UI by going to: <https://localhost:8080>
+    3. Log in using User: admin and Password: from step 6.1
+    4. Navigate to <https://localhost:8080/user-info>
+    5. Click Update Password Button and change to your password of choice
+    6. You will then be logged out, log back in using credentials above
+
+7. Add clusters in Argo
+
+    ``` bash
+    #Connect to api server 
+    argocd login localhost:8080 --username admin --password <same_password_used_in_ui>
+    argocd cluster add k3d-workload-cluster-1 --name workload-cluster-1 --insecure
+    argocd cluster add k3d-workload-cluster-2 --name workload-cluster-2 --insecure
+    argocd cluster add k3d-workload-cluster-3 --name workload-cluster-3 --insecure
+    ```
+
+8. Create applicationset to deploy workloads
+
+    ``` bash
+    kubectl apply -f addon_generator.yaml --insecure-skip-tls-verify
+    ```
+
+9. Navigate to UI by going to: <https://localhost:8080> to see applications being deployed
+
+   > **Note**
+   > At this point all applications are being deployed at once, the dependency configured in our sync waves between prometheus and guestbook is not being respected, this is because in ArgoCD 1.8 the health assesment has been removed from argoproj.io/Application CRD, we will patch this in the next step to enable this healthassesment in order for sync waves to work in our app of apps pattern.
+
+10. Delete applicationset
+
+    ``` bash
+    kubectl delete applicationset addons -n argocd --insecure-skip-tls-verify
+    ```
+
+11. Navigate to UI by going to: <https://localhost:8080> to see all applications will be removed
+
+12. Apply patch to enable health assessment requiring app to be healthy in order to proceed with the next sync wave deployment when using app of apps pattern
+
+    ``` bash
+    kubectl -n argocd patch configmaps argocd-cm --patch-file argocd-cm-patch.yaml --insecure-skip-tls-verify
+    #Restart the argocd server to use the patched configmap
+    kubectl get pods -n argocd --no-headers=true | awk '/argocd-server/{print $1}'| xargs  kubectl delete -n argocd pod --insecure-skip-tls-verify
```

Comment:
Another option here is to delete by label
ex:
```suggestion
    kubectl delete pods -n argocd -l "app.kubernetes.io/name=argocd-server"
```

Reply:
updated to use simpler command